examples/resources/sonarcloud_quality_gate_selection/resource.tf

```diff
@@ -0,0 +1,10 @@
+data "sonarcloud_quality_gate" "awesome_qg" {
+  name = "my_awesome_quality_gate"
+}
+
+data "sonarcloud_projects" "all" {}
+
+resource "sonarcloud_quality_gate_selection" "example_quality_gate_selection" {
+  gate_id   = data.sonarcloud_quality_gate.awesome_qg.gate_id
+  selection = [for project in data.sonarcloud_projects.all : project.key if project.name == "My Awesome Project"]
```

Comment:
The resource does not have a `selection` attribute. There is `project_key`, though I'd rename that to `project_keys`, as it is a set of project keys.